pkg/controller/provider/container/vsphere/collector_test.go

```diff
@@ -0,0 +1,36 @@
+package vsphere
+
+import (
+	liburl "net/url"
+	"testing"
+
+	"github.com/onsi/gomega"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/session"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+func TestTpmCollector(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	collector := Collector{}
+	url, err := liburl.Parse("https://fake.com/sdk")
+	g.Expect(err).To(gomega.BeNil())
+	vimClient := &vim25.Client{
+		Client:         soap.NewClient(url, false),
+		ServiceContent: types.ServiceContent{},
+	}
+	collector.client = &govmomi.Client{
+		SessionManager: session.NewManager(vimClient),
+		Client:         vimClient,
+	}
+	collector.client.ServiceContent.About.ApiVersion = "6.5"
```

Comment:
maybe worth adding name/description for each test so it will be easier to understand what they are testing and desire outcome.

Reply:
i added comments. i think the other option is to split tests (i don't think changing `t.Name` every line is fine).